tests/test_tokenization_common.py

```diff
@@ -666,22 +670,22 @@ def test_added_tokens_do_lower_case(self):
                 text = special_token + " aaaaa bbbbbb low cccccccccdddddddd l " + special_token
                 text2 = special_token + " AAAAA BBBBBB low CCCCCCCCCDDDDDDDD l " + special_token
 
-                new_toks = ["aaaaa bbbbbb", "cccccccccdddddddd", "AAAAA BBBBBB", "CCCCCCCCCDDDDDDDD"]
-
-                toks0 = tokenizer.tokenize(text)  # toks before adding new_toks
+                toks_before_adding = tokenizer.tokenize(text)  # toks before adding new_toks
 
-                added = tokenizer.add_tokens(new_toks)
+                new_toks = ["aaaaa bbbbbb", "cccccccccdddddddd", "AAAAA BBBBBB", "CCCCCCCCCDDDDDDDD"]
+                added = tokenizer.add_tokens([AddedToken(tok, lstrip=True, rstrip=True) for tok in new_toks])
```

Comment:
All added tokens should do `lstrip` and `rstrip` because all `no_split_token` in slow tokenizers strip left and right spaces.

Kindly see line 518~522.

https://github.com/huggingface/transformers/blob/14cc50d081c320331d850a64a54f1d732fa557ea/src/transformers/tokenization_utils.py#L504-L522

Reply:
Thank you so much for the exlications! I agree with you